packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmInstantImpl.kt

```diff
@@ -0,0 +1,44 @@
+package io.realm.internal
+
+import io.realm.RealmInstant
+import io.realm.internal.interop.Timestamp
+
+class RealmInstantImpl(seconds: Long, nanoSeconds: Int) : Timestamp(seconds, nanoSeconds), RealmInstant {
```

Comment:
You could still just use data classes for implementation if you want to piggyback on default `equals` and `hashCode`. But might as well just do the full implementation ... Guess we still want to print the interface name in `toString` anyway, so 🤷

Reply:
data classes doesn't work here when we extend Timestamp...but maybe a reason to introduce that Timestamp as an interface after all 🤔

Reply:
After thinking a bit more...i ended up converting Timestamp to an interface, this allows both `RealmInstantImpl` and `TimestampImpl` to be data classes.